pkg/providers/aws/provider_redis.go

```diff
@@ -131,6 +131,20 @@ func (p *RedisProvider) CreateRedis(ctx context.Context, r *v1alpha1.Redis) (*pr
 		return nil, croType.StatusMessage(errMsg), errorUtil.Wrap(err, errMsg)
 	}
 
+	updatesAllowed, err := resources.VerifyRedisUpdatesAllowed(ctx, p.Client, r.Namespace, r.Name)
+	if err != nil {
+		msg := "failed to verify if updates are allowed"
+		return nil, croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
+	}
+
+	if updatesAllowed {
+		msg, err := p.checkAndApplyMaintenanceUpdates(r, elasticache.New(sess), ec2.New(sess), elasticacheCreateConfig, serviceUpdates)
+		// if there is a status update or error return to controller
+		if err != nil && msg != "" {
```

Comment:
Same as for Postgres - 
seems to me it should look like:
if err != nil {
return nil, statusMsg, err
}
It could be the case with NO error, but with status message, and it considered as error ... Not sure it's ok, as sutus could be also OK/valid, But if not, if status always not valid, so seems to me there is something not clear here with logics. But if we will change it  - it will require the change in checkAndApplyMaintenanceUpdates(...). We could discuss, if my notes are not so clear. Thank you

Reply:
See reasoning above.